lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -3667,131 +3799,26 @@ def _handle_array_type(attype):
                 continue
             table_cols = columns[(schema, row_dict["table_name"])]
 
-            format_type = row_dict["format_type"]
+            coltype = self._reflect_type(
+                row_dict["format_type"],
+                domains,
+                enums,
+                type_description="column '%s'" % row_dict["name"],
+            )
+
             default = row_dict["default"]
+            if isinstance(coltype, DOMAIN) and not default:
```

Comment:
**Federico Caselli** (CaselIT) wrote:

maybe _reflect_type could return `is_domain` too? to avoid the isinstance here and below?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5030

Reply:
Would you prefer a `hasattr` check here and below?

Reply:
it's likely fine as is, just mentioned that since `_reflect_type` knows when it's returning a doman it could return that to the caller